modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserLocalServiceTest.java

```diff
@@ -1122,6 +1448,102 @@ private User _assertLockout(
 		return user;
 	}
 
+	private void _assertUserCreatedWithPasswordPolicy(
+			boolean ldapUser, String password)
+		throws Exception {
+
+		try {
+			User user = _createUser(ldapUser, password);
+
+			Assert.assertEquals(
+				"User was created with incorrect LDAP Server Id",
+				ldapUser ? 1 : -1, user.getLdapServerId());
+			Assert.assertTrue(
+				"During creation, user did not have password policy applied",
+				user.isPasswordReset());
+			Assert.assertNotNull(
+				"User is bypassing portal password policy",
+				user.getPasswordPolicy());
+		}
+		catch (Throwable throwable) {
+			throw new Exception("Unable to create user", throwable);
+		}
```

Comment:
Why wrap this exception? On failure, the stacktrace should be enough to determine the root cause.